hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -443,31 +409,65 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		r.Log.Info("autoscaler rollout pending kubeconfig availability")
 	}
 
-	// Check for readiness and update status
-	var currentCluster capiv1.Cluster
-	if err := r.Get(ctx, client.ObjectKeyFromObject(capiCluster), &currentCluster); err != nil {
-		r.Log.Error(err, "couldn't get CAPI cluster resource", "capiCluster", client.ObjectKeyFromObject(capiCluster))
-		return ctrl.Result{}, err
+	// Compute availability and rollout progress based on the hosted control plane
+	// status
+
+	hcluster.Status.Version = computeClusterVersionStatus(hcluster, hcp)
+
+	r.Log.Info("Successfully reconciled")
+	return ctrl.Result{}, nil
+}
+
+func computeClusterVersionStatus(cluster *hyperv1.HostedCluster, controlPlane *hyperv1.HostedControlPlane) *hyperv1.ClusterVersionStatus {
```

Comment:
nit: we should probably set a variable naming convention for better readability to differentiate between hostedCluster and capi.Cluster, e.g `hyperv1.HostedCluster = hcluster` and `capi.Cluster = cluster`

Reply:
How come we don't get to keep `cluster` for us and make CAPI be `ccluster`?! 😁

Reply:
haha sgtm as well.